lib/reply.js

```diff
@@ -179,10 +179,15 @@ Reply.prototype.send = function (payload) {
   }
 
   if (this[kReplySerializer] !== null) {
-    if (typeof payload !== 'string') {
-      preSerializationHook(this, payload)
-      return this
+    if (typeof payload === 'string') {
+    const contentType = this[kReplyHeaders]['content-type']
+    if (contentType && contentType.startsWith('application/json')) {
+      // Avoid double stringify if already looks like JSON string
+      if (!(payload.startsWith('"') && payload.endsWith('"'))) {
+        payload = JSON.stringify(payload)
+      }
```

Comment:
Not sure about this, is this restoring some functionality?

```suggestion
      if (!(payload[0] === '"' && payload[payload.length - 1] === '"')) {
        payload = JSON.stringify(payload)
      }
```